src/ansi.rs

```diff
@@ -0,0 +1,1913 @@
+//! ANSI Terminal Stream Parsing.
+
+use alloc::borrow::ToOwned;
+use alloc::string::{String, ToString};
+use alloc::vec::Vec;
+
+use core::convert::TryFrom;
+use core::fmt::Write;
+use core::time::Duration;
+use core::{iter, str};
+
+use log::{debug, trace};
+
+use crate::{Params, ParamsIter};
+
+/// Current timestamp provider.
+pub trait TimeProvider: Default {
+    fn now(&self) -> Duration;
+}
+
+#[cfg(not(feature = "no_std"))]
+pub type DefaultTimeProvider = StdTimeProvider;
+
+#[cfg(not(feature = "no_std"))]
+#[derive(Default)]
+pub struct StdTimeProvider;
+
+#[cfg(not(feature = "no_std"))]
+impl TimeProvider for StdTimeProvider {
+    fn now(&self) -> Duration {
+        todo!()
```

Comment:
How should this be handled? Since AFAIK `Instant` is the only way to get monotonic time from the standard library.